WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift

```diff
@@ -1,29 +1,65 @@
 import Foundation
 import UIKit
 import SwiftUI
+import Combine
 
 /// View model for `HubMenu`.
 ///
 final class HubMenuViewModel: ObservableObject {
 
-    let siteID: Int64
+    @Published private(set) var siteID: Int64
 
-    let storeTitle = ServiceLocator.stores.sessionManager.defaultSite?.name ?? Localization.myStore
+    /// The view controller that will be used for presenting the `StorePickerViewController` via `StorePickerCoordinator`
+    ///
+    private(set) unowned var navigationController: UINavigationController?
+
+    var storeTitle: String {
+        ServiceLocator.stores.sessionManager.defaultSite?.name ?? Localization.myStore
+    }
     var storeURL: URL {
         guard let urlString = ServiceLocator.stores.sessionManager.defaultSite?.url, let url = URL(string: urlString) else {
             return WooConstants.URLs.blog.asURL()
         }
-
+        return url
+    }
+    var woocommerceAdminURL: URL {
+        guard let urlString = ServiceLocator.stores.sessionManager.defaultSite?.adminURL, let url = URL(string: urlString) else {
+            return WooConstants.URLs.blog.asURL()
+        }
         return url
     }
 
     /// Child items
     ///
     @Published private(set) var menuElements: [Menu] = []
 
-    init(siteID: Int64) {
+    private var storePickerCoordinator: StorePickerCoordinator?
+
+    private var cancellables = Set<AnyCancellable>()
+
+    init(siteID: Int64, navigationController: UINavigationController? = nil) {
         self.siteID = siteID
+        self.navigationController = navigationController
         menuElements = [.woocommerceAdmin, .viewStore, .reviews]
+        observeSiteForUIUpdates()
+    }
+
+    /// Present the `StorePickerViewController` using the `StorePickerCoordinator`, passing the navigation controller from the entry point.
+    ///
+    func presentSwitchStore() {
+        //TODO-5509: add analytics events
+        if let navigationController = navigationController {
+            storePickerCoordinator = StorePickerCoordinator(navigationController, config: .switchingStores)
```

Comment:
Since `navigationController` is set from the initializer, maybe we can set `storePickerCoordinator` there too? That way we can avoid having to re-creating the coordinator again every time the Switch Store button is tapped 🤔

Reply:
Do you mean passing from outside the `StorePickerCoordinator` (initializing it in `HubMenuCoordinator`) instead of passing the `navigationController`?

Reply:
What I mean is that we can create `storePickerCoordinator` only once in `HubMenuViewModel` initializer using the passed in `navigationController`. WDYT?